presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -17,15 +17,21 @@
 import com.facebook.presto.common.QualifiedObjectName;
 import com.facebook.presto.common.Subfield;
 import com.facebook.presto.common.Subfield.NestedField;
+import com.facebook.presto.common.Subfield.PathElement;
```

Comment:
I'll review this file after you update. It looks like the flow might change if you make some fields in context immutable.